lms/views/dashboard/views.py

```diff
@@ -148,3 +89,17 @@ def _set_lti_user_cookie(self, response):
             max_age=60 * 60 * 24,  # 24 hours, matches the lifetime of the auth_token
         )
         return response
+
+    def _generate_config(self) -> DashboardConfig:
+        config = DashboardConfig(
+            DashboardLinks(
+                assignmentStatsApi=self._to_frontend_template("api.assignment.stats")
+            )
+        )
+        return config
+
+    def _to_frontend_template(self, route_name):
+        route = self.request.registry.introspector.get("routes", "api.assignment.stats")
+        pattern = route["pattern"]
+
+        return re.sub(r"{([^}]+)}", r":\1", pattern)
```

Comment:
Use FE friendly templating :XXX vs {XXX}

Reply:
I wouldn't have a problem with the `{XXX}` approach, even if that's not consistent with how this is done in `h`.

Reply:
While we could work with either without much difficult, a benefit of using `:name` is that this is what the [URLPattern](https://developer.mozilla.org/en-US/docs/Web/API/URL_Pattern_API) API uses. Also we could potentially share code across h and lms.